### ISSUE TITLE
QueryBuilder matches too many types.

### ISSUE THREAD

**Issue by greschd** [aiidateam/aiida_core#817]
The ``QueryBuilder.append`` doesn't match the class name explicitly, but uses a wildcard on the module path. This works for ``Node`` classes which are each defined in a separate module, but fails e.g. for ``aiida.orm.data.base`` types.

Steps to reproduce:
```
from aiida.orm import QueryBuilder
from aiida.orm.data.base import Float

qb = QueryBuilder()
qb.append(Float)
qb.all()
```
This will give instances of all types defined in ``aiida.orm.data.base``, not just ``Float``.

Not sure if this is a duplicate of #273.

**Comment by lekah** [comment]
That is the expected behavior. The problem is that for base types the plugin_type_string convention is violated. The plugin_type_string for Floats should be aiida.orm.data.base.float.Float, and all subclasses should be searchable with by finding types that satisfy aiida.orm.data.base.float.*
@muhrin , @giovannipizzi  This needs to be addressed, in some way. I can't change the behavior of the QB for this, obviously.

For reference:
QueryBuilder().append(Float, subclassing=False) will give you all floats, since it doesn't manipulate the type strings.

**Comment by sphuber** [comment]
I fixed this problem in PR #1192 . Your example should now again only match `Float` nodes
